src/audio/host.c

```diff
@@ -462,6 +462,24 @@ static int host_trigger(struct comp_dev *dev, int cmd)
 						   hd->period_bytes);
 		}
 		break;
+	case COMP_TRIGGER_RELEASE:
+		if (dev->params.direction == SOF_IPC_STREAM_CAPTURE) {
+			struct dma_chan_status status;
+
+			buffer_reset_pos(hd->dma_buffer);
+			dma_status(hd->dma, hd->chan,
+				   &status, SOF_IPC_STREAM_CAPTURE);
+			if (status.r_pos !=
+			    ((uint32_t)hd->dma_buffer->r_ptr & 0x00FFFFFF)) {
```

Comment:
Why do we & 0xffffff here ?

Reply:
because the HOST DMA's DGBRP only has the bit0~bit23 valid.
I will replace it with macro.

Reply:
This is DMA specific, it should be done by dma_status  API on r_pos.